File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/publisher/CustomHeadersClient.kt
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.gcp.pubsub.publisher

// tag::imports[]
import io.micronaut.gcp.pubsub.annotation.PubSubClient
import io.micronaut.gcp.pubsub.annotation.Topic
import io.micronaut.gcp.pubsub.support.Animal
import io.micronaut.messaging.annotation.Header
// end::imports[]

// tag::clazz[]
@PubSubClient
@Header(name = "application-name", value = "petclinic") // <1>
interface CustomHeadersClient {

	@Header(name = "status", value = "healthy") // <2>
	@Topic("animals")
	fun sendWithStaticHeaders(animal: Animal)

	@Topic("animals")
	fun sendWithDynamicHeaders(animal: Animal, @Header(name = "code") code: Int) // <3>

}
// end::clazz[]

File: test-suite-python/src/main/python/micronaut/gcp/pubsub/publisher/CustomHeadersClient.py
from abc import ABC, abstractmethod
from typing import Annotated

# tag::imports[]
from micronaut.gcp.pubsub.annotation import PubSubClient, Topic
from micronaut.gcp.pubsub.support.Animal import Animal
from micronaut.messaging.annotation import MessageHeader
# end::imports[]


# tag::clazz[]
@PubSubClient
@MessageHeader(name="application-name", value="petclinic")  # <1>
class CustomHeadersClient(ABC):

    @MessageHeader(name="status", value="healthy")  # <2>
    @Topic("animals")
    @abstractmethod
    def send_with_static_headers(self, animal: Animal) -> None:
        ...

    @Topic("animals")
    @abstractmethod
    def send_with_dynamic_headers(self, animal: Animal, code: Annotated[int, MessageHeader(name="code")]) -> None:  # <3>
        ...
# end::clazz[]
